metricbeat/module/system/test_system.py

```diff
@@ -388,6 +388,9 @@ def test_process_summary(self):
 
         for evt in output:
             self.assert_fields_are_documented(evt)
+            if evt.get("error", None) is not None:
+                # "error" is a non-fatal error logged by system/process. Ignore and proceed further
+                continue
```

Comment:
[Blocker|Question]
How are you sure it's a _non-fatal error_?

Reply:
if we face any fatal errors, we will not get any events.
We assume that this error is non-fatal and move on.

- If it is indeed non-fatal, the test case should pass with the assertions.
- If it's fatal, the test would fail here
https://github.com/elastic/beats/blob/dd35df4c56e3dc6c0277e039f6dd1d304ade0061/metricbeat/module/system/test_system.py#L395-L396

Reply:
Maybe I should update this comment and be more precise.

Reply:
@AndersonQ, does it look good now?

Reply:
Ah no, 
your concern is valid.
We're missing an important check. I'll update the test for it and let you know.

Reply:
@AndersonQ I've added an assertion that would negate we only receive errors and no metric events. PTAL